### PR TITLE
Update vector layer of data defined buttons after layer selection changes

### LINKED ISSUE #147: Data defined dialog of marker color has wrong feature scope
In the expression builder of the 'marker color' data-defined property the category 'Fields and Values' lists the attributes of the first layer in the layer selection combo box. It should list the attributes of the currently selected layer instead.

### PR DESCRIPTION
**Pull request by SGroe**
Fixes #147

### CODE REVIEW

**Comment by nyalldawson** [comment]
Looks good to me!

File: DataPlotly/gui/plot_settings_widget.py [hunk 1]
```diff
@@ -357,6 +356,17 @@ def getColorDefined(self):
         else:
             self.in_color = hex_to_rgb(self.in_color_combo)
 
+    def refreshLayerSelected(self):
```
Comment (nyalldawson) [inline DataPlotly/gui/plot_settings_widget.py]: We just need a doc string here

**Comment by ghtmtt** [comment]
@nyalldawson can we merge this?

**Comment by nyalldawson** [comment]
Yes

**Comment by ghtmtt** [comment]
thanks @SGroe!

**Pull request merged**